### PR TITLE
Add Niconico support. Let queue limit be adjustable. Add local configs

### PR DESCRIPTION
**Pull request by ATRAN2**
Videos from Niconico can be downloaded and added to the queue now
Queue limit can also be adjusted from local configs. Requires a restart to reload it though.
Also fixed the minor ffmpeg filepath bug

### CODE REVIEW

File: src/renderer/Player.tsx [hunk 1]
```diff
@@ -128,6 +133,11 @@ function Player(props: { mics: InputDevice[] }) {
                 }
                 videoRef.current.play();
                 break;
+              case "NicoQueueItem":
+                setShouldShowPianoRoll(false);
```
Comment (nattofriends) [inline src/renderer/Player.tsx:137]:
```suggestion
                setShouldShowPianoRoll(false);
                setShouldShowAdhocLyrics(false);
```
?

File: src/common/config.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const DEFAULT_CONFIG: KarafriendsConfig = {
+  useLowBitrateUrl: false,
+  predownloadDAM: false,
+  paxSongQueueLimit: 1,
+};
+
+function getConfig(): KarafriendsConfig {
+  // Refer to https://www.electronjs.org/docs/latest/api/app#appgetpathname
+  // for where the config file should be placed. On Windows, it should be %APPDATA%/karafriends/config.yaml
+  const configFilepath: string = path.join(
```
Comment (nattofriends) [inline src/common/config.ts:24]: It's probably some bizarro path on macos, can we print it out?

File: src/renderer/Player.tsx (hunk 1 above)
Comment (ATRAN2) [reply src/renderer/Player.tsx:137]: Yes

File: src/common/config.ts (hunk 2 above)
Comment (ATRAN2) [reply src/common/config.ts:24]: Sure lol

**Review by nattofriends — APPROVED** [review]

**Pull request merged**